htdocs/core/class/commonobject.class.php

```diff
@@ -6350,7 +6350,7 @@ public function updateExtraField($key, $trigger = null, $userused = null)
 			}
 
 			if ($linealreadyfound) {
-				if ($this->array_options["options_".$key] === null) {
+				if ($this->array_options["options_".$key] === null || $this->array_options["options_".$key] === "") {
```

Comment:
Imagine we have an extrafield of type string and we want to set an empty string inside it. We can't anymore.
I think we must separate to case: The case the extrafield is type "link" and the other.
May be we can add this few lines before in the switch that sanitize date:

```
case 'link':
   if ($this->array_options["options_".$key] === "") {
    $this->array_options["options_".$key] = null
  }
```

Can you check/test this way ?

Reply:
@eldy This would work but the behavior is not coherent between the 2 methods, in *insertExtraFields* an empty link is represented by an empty string and with this changes in *updateExtraField* an empty link will be represented by a null value. What do you think?

Reply:
An empty link should be a null value in database (when there is no link) to avoid foreign key errors during insert or update.

Reply:
Okay thanks,

@eldy It's done